packages/material-ui/src/Collapse/Collapse.test.js

```diff
@@ -353,74 +193,133 @@ describe('<Collapse />', () => {
+      const wrapper = mount(
+        <Collapse timeout={timeout} onEntered={next}>
+          <div />
+        </Collapse>,
+      );
+
+      wrapper.setProps({ in: true });
+
       assert.strictEqual(next.callCount, 0);
-      clock.tick(timeout);
+      clock.tick(0);
       assert.strictEqual(next.callCount, 0);
+      clock.tick(timeout);
+      assert.strictEqual(next.callCount, 1);
     });
-  });
 
-  describe('mount', () => {
-    let mountInstance;
+    it('should create proper easeOut animation onEntering', () => {
+      const handleEntering = spy();
 
-    before(() => {
-      mountInstance = mount(<Collapse />)
-        .find('Collapse')
-        .instance();
+      const wrapper = mount(
+        <Collapse
+          onEntering={handleEntering}
+          timeout={{
+            enter: 556,
+          }}
+        >
+          <div />
+        </Collapse>,
+      );
+
+      wrapper.setProps({ in: true });
+      assert.strictEqual(handleEntering.args[0][0].style.transitionDuration, '556ms', true);
     });
 
-    it('instance should have a wrapper property', () => {
-      assert.notStrictEqual(mountInstance.wrapperRef, undefined);
+    it('should create proper sharp animation onExiting', () => {
+      const handleExiting = spy();
+
+      const wrapper = mount(
+        <Collapse
+          {...props}
+          onExiting={handleExiting}
+          timeout={{
+            exit: 446,
+          }}
+        />,
+      );
+
+      wrapper.setProps({
+        in: false,
+      });
+      assert.strictEqual(handleExiting.args[0][0].style.transitionDuration, '446ms', true);
     });
   });
 
   describe('prop: collapsedHeight', () => {
     const collapsedHeight = '10px';
 
     it('should work when closed', () => {
-      const wrapper = shallow(<Collapse {...props} collapsedHeight={collapsedHeight} />);
-      const child = new ReactWrapper(
-        wrapper
-          .dive()
-          .props()
-          .children('entered'),
-      );
+      const wrapper = mount(<Collapse {...props} collapsedHeight={collapsedHeight} />);
+      const child = wrapper.find('Transition').childAt(0);
       assert.strictEqual(child.props().style.minHeight, collapsedHeight);
     });
 
     it('should be taken into account in handleExiting', () => {
-      const wrapper = shallow(<Collapse {...props} collapsedHeight={collapsedHeight} />);
-      const instance = wrapper.dive().instance();
-      const element = { style: { height: 666, transitionDuration: undefined } };
-      instance.handleExiting(element);
-      assert.strictEqual(element.style.height, collapsedHeight, 'should have 0px height');
+      const handleExiting = spy();
+      const wrapper = mount(
+        <Collapse {...props} collapsedHeight={collapsedHeight} onExiting={handleExiting} />,
+      );
+      wrapper.setProps({ in: false });
+
+      assert.strictEqual(
+        handleExiting.args[0][0].style.height,
+        collapsedHeight,
+        'should have 0px height',
```

Comment:
Weird.

Reply:
Mocking height is difficult 😅

Reply:
I would expect the new tests to work...

Reply:
Looks like args are passed by reference so checking the value like this won't work

Reply:
To be more specific `collapsedHeight = 10px` while the description says: 0px.